3rdparty/ndsrvp/include/cvutils.hpp

```diff
@@ -0,0 +1,95 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.	
+
+#ifndef OPENCV_NDSRVP_CVUTILS_HPP
+#define OPENCV_NDSRVP_CVUTILS_HPP
+
+#include <cstring>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <climits>
+
+namespace cv {
+
+namespace ndsrvp {
+
+void* fastMalloc(size_t size);
```

Comment:
I have concerns regarding possible name overlap with existing OpenCV functions. Can we "hide" this header-file in the `src/` folder.

Reply:
Done.

Reply:
Sorry, I haven't made myself clear enough. Idea is to leave as little stuff as possible in "ndsrvp_hal.hpp" (taking in account included files) - we see this file as a public interface of the HAL library. This file will be included into parts of OpenCV library and any extra function or constant defined there can potentially overlap with ones in OpenCV source code. Ideally this interface should contain only necessary function declarations and `cv_hal_*` macro overrides.

By "hiding cvutils.hpp" I meant removing the file from mentioned public interface (`ndsrvp_hal.hpp` and included headers) and including it explicitly in source files which need it. In OpenCV library we usually move such files to the `src/` folder of the module and our build/install procedure makes it impossible to include them into any of the public headers.

Reply:
The inclusion is moved into source files. Public interface is independent to `cvutils.hpp` and `cvutils.cpp`, meanwhile `cvutils` also have become independent to any other OpenCV source code.

Reply:
Currently there are compilation issues in cvutils.cpp. It requires some extra headers:
```
/work/opencv/3rdparty/ndsrvp/src/cvutils.hpp:55:8: error: 'int32x2_t' does not name a type; did you mean 'int32_t'?
   55 | inline int32x2_t vclip(int32x2_t x, int32x2_t a, int32x2_t b)
      |        ^~~~~~~~~
      |        int32_t
/work/opencv/3rdparty/ndsrvp/src/cvutils.hpp:66:19: error: 'uchar' does not name a type; did you mean 'u_char'?
   66 | template<> inline uchar saturate_cast<uchar>(int v)     { return __nds__uclip32(v, 8); }
      |                   ^~~~~
      |                   u_char
...
```

Reply:
Fixed. Core module HAL interface is required by `cvutils` now.